packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -143,29 +143,13 @@ function handleUnenvAliasedPackages(
 		};
 	});
 
-	build.initialOptions.banner = { js: "", ...build.initialOptions.banner };
-	build.initialOptions.banner.js += dedent`
-		function __cf_cjs(esm) {
-		  const cjs = 'default' in esm ? esm.default : {};
-			for (const [k, v] of Object.entries(esm)) {
-				if (k !== 'default') {
-					Object.defineProperty(cjs, k, {
-						enumerable: true,
-						value: v,
-					});
-				}
-			}
-			return cjs;
-		}
-		`;
-
 	build.onLoad(
 		{ filter: /.*/, namespace: REQUIRED_UNENV_ALIAS_NAMESPACE },
 		({ path }) => {
 			return {
 				contents: dedent`
 					import * as esm from '${path}';
-					module.exports = __cf_cjs(esm);
+					module.exports = Object.assign(esm.default || {}, ...Object.entries(esm).filter(([k,]) => k !== 'default'));
```

Comment:
Did we use `defineProperty` for a reason? Is it possible that the `default` property object is locked down from being written to?

Reply:
Good catch, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#description

```ts
module.exports = Object
  .defineProperties(
    esm.default || {}, 
    Object.fromEntries(
      Object.entries(esm)
         .filter(([k,]) => k !== 'default'))
         .map(([k, value]) => [k, {value, enumerable: true}])
    )    
  );

Reply:
I fixed the exports assigment to use Object.defineProperty to ensure that the properties are enumerable.